src/DDserver.cpp

```diff
@@ -353,51 +360,48 @@ namespace DoubleD
         }
     }
 
-    std::string DDserver::m_handleRequest(const unsigned int TIMEOUT, std::string lockName, const double LIFETIME)
-    {
-        auto startTime = std::chrono::high_resolution_clock::now();
-        auto currentTime = std::chrono::high_resolution_clock::now();
-        std::chrono::duration<double> difference = currentTime - startTime;
-        do
-        {
-            DDserver::m_storageMutex.lock();
-            
-            if (DDserver::m_lockVector.size() == 0)
-            {
-                Lock lock(lockName, LIFETIME);
-                DDserver::m_lockVector.push_back(lock);
-                DDserver::m_storageMutex.unlock();
-                return lock.m_getSessionToken();
-            }
-
-            for (long unsigned int i = 0; i < DDserver::m_lockVector.size(); i++)
+    // returns a Lock if a Lock can be acquired, otherwise returns boost::none.
+    boost::optional<Lock> DDserver::m_getLock(std::string lockName, const double LIFETIME){
```

Comment:
Since you're a boost fanboy you may like the usage of `optional` here! Now we really only need to call lock() and unlock() once!